updated membership list

0ced0 opened the pull request:
## Description
<!-- What does this PR do? Why is it needed? -->

## Type of Change
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation
- [ ] Refactoring

## Quick Checklist
- [ ] Code tested locally
- [ ] No warnings or errors
- [ ] Follows project structure (`backend/apps/`, `frontend/src/`)
- [ ] No `__pycache__`, `.env`, or `node_modules` committed
- [ ] Database migrations tested (if applicable)
- [ ] Updated documentation (if needed)

## For Major Changes Only
<details>
<summary>Click if this is a significant feature or breaking change</summary>

### Implementation Details
- Files changed:
- PRD section implemented:
- Completion: %

### Potential Issues
- What could break:

### Testing Done
- [ ] Tested on development
- [ ] Tested edge cases
- [ ] No existing functionality broken

</details>

## Screenshots
<!-- Add screenshots for UI changes -->

---
**⚠️ Critical Rules:**
- No commits to root directory (except docs/)
- No hard-coded credentials
- All apps in `backend/apps/`

**📝 Note:** @emperuna reviews all changes

0ced0 committed:
updated membership list

emperuna reviewed (commented):
# Code Review Feedback - MembershipListPage Refactor

**Reviewer:** emperuna  
**Date:** November 9, 2025  
**Branch:** features/members  
**Status:** ❌ **Changes Requested - Cannot Merge**

---

## Executive Summary

Aside sa slightly UI changes and at mga dinagdag mong components, halos nasira na yung functions ng iba lalo sa **CRUD operation** (only yung delete lang ang nagana). As for the filtering yung ministry mo ang **‘di nagana bukod pa ro’n naka hardcoded yung gawa mo instead of dynamic.**

**Issues Found:**
- Broken archive button, pag cinonfirm mo hindi nagana.
- Hindi align yung gawa mo sa members table as you can see ‘di pantay
- Hardcoded colors everywhere, ‘di nasusunod yung sa theme store nasa theme.store.js at App.css

---

## Critical Issues (Must Fix Before Merge)

### 1. 🔴 Missing Imports - Application Crashes

**File:** `MembershipListPage.jsx` (Lines 62-63)
```jsx
await membersApi.archiveMember(archiveState.member.id); 
// ❌ ERROR: membersApi is not imported
```

**File:** `ListCards.jsx` (Restore button)
```jsx
<HiArrowPath className="..." />
// ❌ ERROR: HiArrowPath is not imported from 'react-icons/hi2'
```

**Impact:**  
Application will **crash** when users try to:
- Archive a member
- Restore an archived member

**Fix Required:**
```jsx
// Add to MembershipListPage.jsx
import { membersApi } from '../../../api/members.api';

// Add to ListCards.jsx
import { HiArrowPath } from 'react-icons/hi2';
```

---

### 2. 🔴 Broken Layout Architecture

**File:** `ListHeaders.jsx` & `ListCards.jsx`

**Problem:**
```jsx
// Current implementation (BROKEN):
className={`${collapsed ? 'mr-[18.7%]' : 'mr-[12.6%]'} transition-all`}
```

**Why This Fails:**

| Issue | Description | Impact |
|-------|-------------|---------|
| **Magic Numbers** | Hardcoded percentages (18.7%, 12.6%, 9%, etc.) with no documentation | Impossible to maintain |
| **Wrong CSS Property** | Uses `margin-right` instead of `width` | Alignment breaks easily |
| **Screen Size Dependent** | Only works on specific screen resolution | Fails on 1366px, 2560px, mobile |
| **Content Dependent** | Adding longer names/text breaks layout | Not scalable |
| **Per-Column Values** | Each column has different margin calculations | Cascading layout issues |

**Test Cases That Fail:**
```
✅ Developer's 1920px MacBook screen: Works
❌ User with 1366px laptop: Columns misaligned
❌ User with 2560px 4K monitor: Huge gaps
❌ Member name "Christopher Emmanuel Johnson III": Breaks layout
❌ Long ministry name added to database: Columns shift
```

---

### 3. 🟡 Hardcoded Data Instead of API

**File:** `MemberListNavBar.jsx` (Lines 260-290)

**Problem:**
```jsx
// Ministry filter dropdown
<button onClick={() => handleSelectMinistry('media')}>Media</button>
<button onClick={() => handleSelectMinistry('worship')}>Worship</button>
<button onClick={() => handleSelectMinistry('music')}>Music</button>
<button onClick={() => handleSelectMinistry('youth')}>Youth</button>
```

**Issues:**
- ❌ Ministries are **hardcoded** in the component
- ❌ Will break when new ministries are added to database
- ❌ Won't reflect renamed/deleted ministries
- ❌ Ignores the `ministries` data already fetched from API (line 24)

**You Already Have the Data:**
```jsx
const { ministries, loading: ministriesLoading } = useMinistries(); // Line 24
// But you're not using it! 🤔
```

---

### 4. 🟡 Over-Coupling with Sidebar Context

**Problem:** Three separate components all depend on `useSideBar()`:

```jsx
// MemberListNavBar.jsx
const { user } = useAuth();
const canManage = MANAGER_ROLES.includes(user?.role);
// ❌ Imports useSideBar but doesn't use it!

// ListHeaders.jsx
const { collapsed, toggleSidebar } = useSideBar();
// ✓ Uses collapsed state for responsive margins

// ListCards.jsx
const { collapsed, toggleSideBar } = useSideBar();
// ✓ Uses collapsed state for responsive widths
```

**Issues:**
- **Unnecessary Re-renders:** All 3 components re-render when sidebar toggles
- **Tight Coupling:** Components can't be reused outside sidebar context
- **Wrong Responsibility:** Table headers shouldn't know about sidebar state
- **Performance:** Sidebar state change triggers 3 component updates

**Better Approach:**  
Use CSS Grid/Flexbox for responsive layout. No JavaScript state needed.

---

### 5. 🟡 Dead Code & Poor Cleanup

**File:** `MemberListNavBar.jsx` (Lines 102-111)

```jsx
// const handleSelectGender = (value) => {
//     setGenderFilters(prev => ({ ...prev, gender: value }));
//     setGenderIsOpen(false);
// };

// const handleSelectMinistry = (value) => {
//     setMinistryFilters(prev => ({ ...prev, ministry: value }));
//     setMinistryIsOpen(false);
// };
```

**Issues:**
- ❌ Commented-out duplicate code left in production
- ❌ Shows rushed work without proper cleanup
- ❌ Confusing for future developers

**Fix:** Delete commented code entirely.

---

## What You Did Right ✅

### Good UX Ideas:
- ✅ **Unified navbar** - Putting search, filters, and actions together is cleaner
- ✅ **CSV import integration** - Good addition to the navbar
- ✅ **Filter dropdowns** - More intuitive than separate filter components
- ✅ **Visual consistency** - Maintained existing design language and color scheme

**Goods tayo dito** pwede siyang i-keep. Ang problema ay yung **implementation**, not the concept.

---

### Must Fix (Critical):

1. **Fix Application Crashes:**
   ```jsx
   // Add missing imports
   import { membersApi } from '../../../api/members.api';
   import { HiArrowPath } from 'react-icons/hi2';
   ```

2. **Fix Layout System:**
   - **Option A:** Revert to original percentage-based widths (`w-[20%]`, etc.)
   - **Option B:** Implement proper CSS Grid (see code example above)
   - **Option C:** Use Flexbox with flex fractions
   - ❌ **Do NOT use:** Sidebar-dependent margin calculations

3. **Use Dynamic Ministry Data:**
   ```jsx
   {ministries.map(ministry => (
     <button onClick={() => handleSelectMinistry(ministry.id)}>
       {ministry.name}
     </button>
   ))}
   ```

### Should Fix (Quality):

4. **Remove Sidebar Coupling:**
   - Remove `useSideBar()` from components that don't need it
   - Use CSS for responsive layout instead of JS state

5. **Clean Up Code:**
   - Delete all commented-out code
   - Remove unused imports

---

## Key Takeaway

The current implementation introduces **critical bugs** and a **fragile layout system** that breaks easily. The original code was:

**Before (Old Code):**
- ✅ Stable layout
- ✅ Responsive across screen sizes
- ✅ Easy to maintain
- ✅ Uses dynamic data from API

**After (New Code):**
- ❌ 2 critical crashes
- ❌ Fragile responsive behavior
- ❌ Only works on tested screen size
- ❌ Hard to maintain

Si Marc na bahala kung i-aapprove niya ba 'to or hindi since siya naman ang nag l-lead sa'yo sa frontend. I gave my reviews already tignan mo nalang second opinion niya sa gawa mo.

---

0ced0 committed:
fixed archive functionality
fixed restore functionality
fixed ministry filter
fixed overflow handing (need to use width)
removed sidebar from memberlistnavbar
cleaned up dead code
added new layout for smaller screens
